bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java

```diff
@@ -89,6 +99,22 @@ public String getThingID() {
         return this.thingID;
     }
 
+    public Configuration getThingConfig() {
+        return getConfig();
+    }
+
+    public DiscoveryServiceRegistry getDiscoveryServiceRegistry() {
```

Comment:
I didn't load this into my IDE yet, but where is this method used?

Reply:
It's in the PhilipsTV PR but will also likely be used when I do the Sony Scalar addition shortly as well.  I wanted to get it in now so I didn't have it sitting in both.  This PR provides a few of those small things.  It's called in PR #15352 go to PhilipsTVConnectionManager line 142.

Reply:
It's in the PhilipsTV PR but will also likely be used when I do the Sony Scalar addition shortly as well. I wanted to get it in now so I didn't have it sitting in both. This PR provides a few of those small things. It's called in PR https://github.com/openhab/openhab-addons/pull/15352 go to PhilipsTVConnectionManager line 142.

Reply:
I'm not sure if you will actually need it. You could have a look at #16250 although I'm not sure if it's a similar use-case. Maybe you find inspiration, maybe not. I cannot fully assess it right now, since it would require a deeper look at that PR. 

In any case it would be better to remove it from this PR since it's still unused. If it will ultimately be needed, you can introduce it with the first PR requiring it.

Reply:
Just to be clear, are you wanting me to pull just this function or the whole thing out of the factory and all?  I don't have a problem with either, just making sure I understand the request.

Reply:
Yes, I would leave everything out until it will eventually be needed.

It was a good idea to extract the bugfixes from #15352 so they can be merged faster. You could even have extracted them individually, and it could have been even smoother, and each one could be mentioned individually in the release notes generated from the list of PR's. 😉

It has pros and cons, of course, but usually I think the pros outweights the cons. When increments are small, the risk of conflicts is usually also smaller. And since the reviews and merges will usually also be faster, everything can be rebased gradually, and there will be fewer and fewer still pending differences to manage. Additionally, it will be easier (well, possible) to cherry-pick commits for backporting specific bugfixes into patch releases.